homeassistant/components/goalzero/switch.py

```diff
@@ -0,0 +1,153 @@
+"""Support for Goal Zero Yeti."""
+
+import logging
+
+from requests import post, get, exceptions
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_HOST, STATE_OFF, STATE_ON
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import ToggleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.matches_regex(
+            r"\A(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2 \
+            [0-4][0-9]|[01]?[0-9][0-9]?)\Z"
+        ),
+        vol.Optional(CONF_NAME, default="Yeti"): cv.string,
+    }
+)
+
+header = {
+    "Content-Type": "application/json",
+    "User-Agent": "YetiApp/1340 CFNetwork/1125.2 Darwin/19.4.0",
+    "Connection": "keep-alive",
+    "Accept": "application/json",
+    "Accept-Language": "en-us",
+    "Content-Length": "19",
+    "Accept-Encoding": "gzip, deflate",
+    "Cache-Control": "no-cache",
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the aREST switches."""
+
+    name = config[CONF_NAME]
+    host = config[CONF_HOST]
+
+    try:
+        response = get("http://" + host + "/state", timeout=10).json()
```

Comment:
Please move this code into a PyPi package

https://developers.home-assistant.io/docs/api_lib_index/

Reply:
I'll do my best

Reply:
@bdraco Done